pilot/pkg/model/push_context.go

```diff
@@ -804,6 +809,21 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of service account aliases in the registry
+func (ps *PushContext) initServiceAccountAliases(env *Environment, services []*Service) {
+	for _, svc := range services {
+		if ps.ServiceAccountAliases[svc.Hostname] == nil {
+			ps.ServiceAccountAliases[svc.Hostname] = map[int][]string{}
+		}
+		for _, port := range svc.Ports {
+			if port.Protocol == protocol.UDP {
+				continue
+			}
+			ps.ServiceAccountAliases[svc.Hostname][port.Port] = env.GetIstioServiceAccountAliases(svc, []int{port.Port})
+		}
```

Comment:
I thikn you can get rid of this function. You can also get rid of the GetIstiServiceAccountAliases. Instead, GetIstioServiceAccounts should return all the service accounts (originals and aliases) because we are using all of them in the exact same way - for san validation in envoy cluster. It does not make sense to make two backend store calls for this.

Reply:
Resolved, same as above. Thanks!